modules/builds-identifying-image-change-triggers.adoc

```diff
@@ -0,0 +1,70 @@
+// Module included in the following assemblies:
+//
+// * builds/triggering-builds-build-hooks.adoc
+
+[id="builds-image-change-trigger-identification_{context}"]
+= Image change trigger identification
+
+As a developer, if you have configured image change triggers, you can identify which image change initiated the last build.
+
+To accomplish this, you must identify elements in your build configuration's specification and status that are related to image change triggers.
+
+This way, you can use the timestamp in `buildConfig.status.imageChangeTriggers` to identify the most recent build. Then you can use the name and namespace values of the image stream that triggered this build to find the corresponding image change trigger in the `buildConfig.spec.triggers`.
+
+
+== Image change trigger elements in the specification
+
+In your build configuration specification, `buildConfig.spec.triggers` is an array of build trigger policies, `BuildTriggerPolicy`.
+
+Each `BuildTriggerPolicy` has a `type` field and set of pointers. The `type` can be set to only one pointer. For image change triggers, the `type` can take different forms, depending on how you defined the triggers:
+
+* `buildConfig.spec.triggers[i].kind = ImageChange`
+* `buildConfig.spec.triggers[i].imageChange`, deprecated, whose value is the JSON or YAML of an `ImageChangeTrigger`
```

Comment:
Replace this with:

Then, the `imageChange` field is the pointer to a `ImageChangeTrigger` object.  So this will be set.  It has 3 fields:

- `lastTriggeredImageID`:  This field is deprecated as of 4.8.  It is still being set in 4.8, bu that will change in future releases.  It contains the resolved image reference for the `ImageStreamTag` when the last build was triggered from this `BuildConfig`
- `paused`:  This field is used to temporarily disable this particular image change trigger
- `from`: this field is used to reference the `ImageStreamTag` that drives this image change trigger.  Its type is the core Kubernetes type `OwnerReference`.  Hence, `from` has the following fields of note:
- - `kind`: in this case, the only supported value is `ImageStreamTag`
- - `namespace`: the namespace where the `ImageStreamTag` lives
- - `name`: the name of the `ImageStreamTag`

Reply:
Updated, thanks.